sdks/java/core/src/main/java/org/apache/beam/sdk/coders/GeneratedRowCoder.java

```diff
@@ -0,0 +1,316 @@
+
+  // Cache for Coder class that are already generated.
+  private static Map<UUID, Coder<Row>> generatedCoders = Maps.newHashMap();
+
+  static {
+    // Initialize the CODER_MAP with the StackManipulations to create the primitive coders.
+    // Assumes that each class contains a static of() constructor method.
+    CODER_MAP = Maps.newHashMap();
+    for (Map.Entry<TypeName, Coder> entry : RowCoder.CODER_MAP.entrySet()) {
+      StackManipulation stackManipulation = MethodInvocation.invoke(
+          new ForLoadedType(entry.getValue().getClass()).getDeclaredMethods()
+              .filter(ElementMatchers.named(("of")))
+              .getOnly());
+      CODER_MAP.putIfAbsent(entry.getKey(), stackManipulation);
+    }
+  }
+
+  public static Coder<Row> of(Schema schema, UUID coderId) {
+    return generatedCoders.computeIfAbsent(coderId,
+        h -> {
+          TypeDescription.Generic coderType =
+              TypeDescription.Generic.Builder.parameterizedType(Coder.class, Row.class).build();
+          DynamicType.Builder<Coder> builder =
+              (DynamicType.Builder<Coder>) BYTE_BUDDY.subclass(coderType);
+          builder = createComponentCoders(schema, builder);
+          builder = implementMethods(builder, schema);
+          try {
+            return builder
+                .make()
+                .load(
+                    Coder.class.getClassLoader(),
+                    ClassLoadingStrategy.Default.INJECTION)
+                .getLoaded()
+                .getDeclaredConstructor().newInstance();
+          } catch (InstantiationException | IllegalAccessException
+              | NoSuchMethodException | InvocationTargetException e) {
+            throw new RuntimeException("Unable to generate coder for schema " + schema);
+          }
+        });
+  }
+
+  static DynamicType.Builder<Coder> implementMethods(
+      DynamicType.Builder<Coder> builder, Schema schema) {
+    return builder
+        .defineMethod("getSchema", Schema.class, Visibility.PRIVATE, Ownership.STATIC)
+        .intercept(FixedValue.reference(schema))
+        .method(ElementMatchers.named("encode"))
+        .intercept(new EncodeInstruction())
+        .method(ElementMatchers.named("decode"))
+        .intercept(new DecodeInstruction());
+  }
+
+  private static class EncodeInstruction implements Implementation {
+    static final ForLoadedType LOADED_TYPE = new ForLoadedType(EncodeInstruction.class);
+    @Override
+    public ByteCodeAppender appender(Target implementationTarget) {
+      return (methodVisitor, implementationContext, instrumentedMethod) -> {
+        StackManipulation manipulation = new StackManipulation.Compound(
+            // Array of coders.
+            FieldAccess.forField(implementationContext
+                .getInstrumentedType()
+                .getDeclaredFields()
+                .filter(ElementMatchers.named(CODERS_FIELD_NAME))
+                .getOnly()).read(),
+            // Element to encode.
```

Comment:
add comment: skipping index 0 because index 0 is reserved for "this"

Reply:
done